nixos/lib/test-driver/Machine.pm

```diff
@@ -250,7 +250,8 @@ sub connect {
         $self->start;
 
         local $SIG{ALRM} = sub { die "timed out waiting for the VM to connect\n"; };
-        alarm 300;
+        # 50 minutes
+        alarm 3000;
```

Comment:
Maybe add to your comment that it should probably be reverted, and is added due to #49384